From ac8e59f67a2659765ec4d6a4f26b0553a4ae7054 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 10 Mar 2025 18:06:03 +0100
Subject: [PATCH 1657/1695] product/optee-stm32mp2: fix default latency value

Fix default OPP change latency value to 1000uS.
Zero is treated as infinite value by linux SCMI driver:
cpu/cpu0/cpufreq/cpuinfo_transition_latency:4294967295

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 74ea886e55d488f784f60492a6d0c6e5ab1f5e6c product/optee-stm32mp2: fix default latency value]
Change-Id: I54144d01db5f6c6eca0390cb8b6d884f5a9cc357
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/443251
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Valentin CARON <valentin.caron@foss.st.com>
---
 .../SCP-firmware/product/optee-stm32mp2/fw/config_all.c         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c
index a11e54112..bdc575da7 100644
--- a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c
+++ b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c
@@ -923,7 +923,7 @@ static void set_resources(struct scpfw_config *cfg)
                         (fwk_id_t)FWK_ID_ELEMENT_INIT(FWK_MODULE_IDX_PSU, psu_index);
                     dvfs_data[dvfs_index].clock_id =
                         (fwk_id_t)FWK_ID_ELEMENT_INIT(FWK_MODULE_IDX_CLOCK, clock_index);
-                    dvfs_data[dvfs_index].latency = 0; /* not set, used for async access */
+                    dvfs_data[dvfs_index].latency = 1000;
                     dvfs_data[dvfs_index].sustained_idx = perfd_cfg->initial_opp;
                     dvfs_data[dvfs_index].opps =
                         fwk_mm_calloc(perfd_cfg->dvfs_opp_count,
-- 
2.47.2

